### PR TITLE
Document hybrid apps with SDK v2

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Use SDK v2 in hybrid apps support document.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [ ] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

File: docs/hybrid_applications.md [hunk 1]
```diff
@@ -14,7 +16,7 @@ A few constraints to keep in mind:
     - Through _auto-instrumentation_ - Some React classes and methods are modified to automate this.
     - Through _manual instrumentation_ - For example, if you want to report something you consider an error but that is not going to crash the app.
       Auto-instrumentation for JavaScript errors, resources, and interactions can only be started from JavaScript code.
-2. You cannot initialize more than one instance of the native `DatadogSDK` because it uses a singleton pattern on the native side. That means that once you initialize the native SDK on the native side or on the ReactNative side (by calling `DdSdkReactNative.initialize`) it is initialized for both.
+2. You can share the same instance of the core SDK between native and React Native. That way once you initialize the native SDK on the native side or on the React Native side (by calling `DdSdkReactNative.initialize`) it is initialized for both and events will appear in the same RUM Session. React Native uses the default core instance.
```
Comment (buraizu) [inline docs/hybrid_applications.md]:
```suggestion
2. You can share the same instance of the core SDK between native and React Native. This allows you to initialize the native SDK on either the native side or on the React Native side (by calling `DdSdkReactNative.initialize`) and have it initialized for both sides, with events appearing in the same RUM session. React Native uses the default core instance.
```

File: docs/hybrid_applications.md [hunk 2]
```diff
@@ -29,18 +31,26 @@ This also initializes the SDK on the native side. You are able to call both the
...
     // The version will be set by @datadog/mobile-react-native
-    implementation "com.datadoghq:dd-sdk-android"
+    implementation "com.datadoghq:dd-sdk-android-rum"
+    implementation "com.datadoghq:dd-sdk-android-logs"
+    implementation "com.datadoghq:dd-sdk-android-trace"
+    implementation "com.datadoghq:dd-sdk-android-webview"
 ```
 
-On iOS, add the Datadog iOS SDK to your dependencies in your `ios/Podfile` to use it in Objective C files:
+On iOS, add the Datadog iOS SDKs to your dependencies in your `ios/Podfile` to use it in Objective C files:
```
Comment (buraizu) [inline docs/hybrid_applications.md]:
```suggestion
On iOS, add the Datadog iOS SDKs to your dependencies in your `ios/Podfile` to use in Objective C files:
```

File: docs/hybrid_applications.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Hybrid applications
 
+Note: This document uses SDK v2. For v1 documentation, refer to the [previous version of this document][10].
```
Comment (buraizu) [inline docs/hybrid_applications.md]:
```suggestion
Note: This document uses SDK v2. For v1 documentation, see the [previous version of this document][10].
```

**Review by buraizu — DISMISSED** [review]
Thanks, approving with some minor feedback and clarification

**Review by 0xnm — DISMISSED** [review]

**Review by jonathanmos — DISMISSED** [review]

**Review by 0xnm — APPROVED** [review]

**Pull request merged**